zebra/zebra_rib.c

```diff
@@ -3255,12 +3256,26 @@ static int rib_meta_queue_add(struct meta_queue *mq, void *data)
 		return -1;
 
 	/* Invariant: at this point we always have rn->info set. */
-	if (CHECK_FLAG(rib_dest_from_rnode(rn)->flags,
-		       RIB_ROUTE_QUEUED(qindex))) {
-		if (IS_ZEBRA_DEBUG_RIB_DETAILED)
+	/* A route node must only be in one sub-queue at a time. */
+	curr_qindex_bitmask = CHECK_FLAG(rib_dest_from_rnode(rn)->flags, MQ_BIT_MASK);
+	if (curr_qindex_bitmask) {
```

Comment:
let's keep this the same as much as possible

`if (CHECK_FLAG(rib_dest_from_rnode(rn)->flags, MQ_BIT_MASK) {`

Reply:
> let's keep this the same as much as possible
> 
> `if (CHECK_FLAG(rib_dest_from_rnode(rn)->flags, MQ_BIT_MASK) {`

I have made changes. These are not very important, IMHO. For example, if we use gcc with a -02 optimization level, both code variants are the same in disassembling output. curr_qindex_bitmask stores in %edx register.

```
0x000055555564aee2 <+178>:   and    $0x7ff,%edx
...
0x000055555564b013 <+483>:   imul   $0x77cb531,%edx,%eax
```

Reply:
It wasn't about the compiler, it was about readability and consistency of the code with other parts of the code base.